### PR TITLE
Fix bug in outer_indexer where the special case of an empty right array resulted in bogus return data.

### PR DESCRIPTION
**Pull request by agijsberts**
Fixes bug described in issue #10618.

### CODE REVIEW

**Comment by jreback** [comment]
tests pls

**Comment by jreback** [comment]
pls add a release note in whatsnew/0.17.0

any tests at a higher level that hit this code?

**Comment by agijsberts** [comment]
As far as I can tell this boundary case is already tested at the higher level (https://github.com/pydata/pandas/blob/master/pandas/tests/test_index.py#L919). Those tests never hit the low level cython code though, since these boundary cases are preempted in the high-level code (e.g., https://github.com/pydata/pandas/blob/master/pandas/core/index.py#L1419).

File: pandas/tests/test_algos.py [hunk 1]
```diff
@@ -168,6 +168,38 @@ def _test_vector_resize(htable, uniques, dtype, nvals):
...
+        for dtype, indexer in typemap:
+            left = np.arange(3, dtype = dtype)
+            right = np.arange(2,5, dtype = dtype)
+            empty = np.array([], dtype = dtype)
+
+            result, lindexer, rindexer = indexer(left, right)
+            tm.assertIsInstance(result, np.ndarray)
+            tm.assertIsInstance(lindexer, np.ndarray)
+            tm.assertIsInstance(rindexer, np.ndarray)
+            np.testing.assert_equal(result, np.arange(5, dtype = dtype))
```
Comment (jreback) [inline pandas/tests/test_algos.py]: use ``tm.assert_numpy_array_equal``

**Comment by jreback** [comment]
once you fix the testing function, pls squash, and ping.

**Comment by agijsberts** [comment]
Done.

**Comment by jreback** [comment]
thanks you sir!

**Pull request merged**